compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -70,7 +72,9 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
         current_function: Option<FuncId>,
     ) -> Self {
         let bound_generics = Vec::new();
-        Self { elaborator, crate_id, current_function, bound_generics, in_loop: false }
+        let in_loop = false;
+        let call_stack = Vec::new();
```

Comment:
I think this will result in the call stack being reset if we ever elaborate within the interpreter, than that elaborator also runs interpreted code.